### DISCUSSION TITLE
Extending "php artisan about" output

### DISCUSSION THREAD

**Discussion by roberto-butti** [laravel/framework#45853] [Ideas] [answered]
I suggest adding more information to the `php artisan about` command.
I can open a PR if you think that this feature could have sense and provide any value.
To keep the about output command minimal, I would like to suggest enabling showing additional information with a specific parameter like `--extended`
The idea is:
- add new option `--extended` option
- add and show information about .env configurations for DB
- add information about path and directories used by Laravel (lang, public, storage, resource, cache etc)
- add information about some relevant php.ini parameters (like max upload etc)

Let me know what you think, if this potential PR could be useful (or not) and which kind of information you would like to see.

I have attached some examples of the output:
![Ouput for database configuration](https://pbs.twimg.com/media/FnoHwJVXkAI-COw?format=jpg&name=small)
![extended Ouput](https://pbs.twimg.com/media/FnoNwscXgAAGsm-?format=jpg&name=small)
![information about operating system](https://pbs.twimg.com/media/FnoQvuFXoAAuISx?format=jpg&name=small)

**Comment by henzeb** [comment]
I never look in the about. For database information we already have db:show. (although that one could have a minimal flag as a large database with a lot of tables is snowing the real details away).

You could also try and release a package and see how much people are interested in it?

**Reply by roberto-butti** [reply]
Thank you, @henzeb, for the valuable feedback.
In the past, even before the existence of the `php artisan about` command and the `php artisan db:show` command, I had released an open-source package for inspecting and checking some configuration parameters and values.
With the creation of these two commands in the framework core, which partly overlap with the package's functionality, I suggest bringing some of the package's functionality (such as those listed above) into the about command.
And yes, leaving the `php artisan about` command "minimal" and delegating more in-depth inspections to the external package is an option.
But since I consider the about command (and db:show) to be very useful and valuable, I was wondering about contributing to their extension.

**Comment by marco-introini** [comment]
In my opinion this is a great idea. 
It can be very useful also in production to compare with local dev environment

**Reply by roberto-butti** [reply]
Which kind of information do you expect to be shown by about command (as extended information and not by default)

**Reply by marco-introini** [reply]
I'd like to have some info about php.ini (for example: max upload size and memory limit)

**Reply by roberto-butti** [reply]
In this case, the php.ini loaded by the PHP CLI could be different from the PHP fpm. So if you run `php artisan about` from the command line, some runtime configurations could be different from the configurations loaded by the process used to serve the web part. Just a note as a reminder. In my php package for showing the runtime i also have the feature to run the "inspector" also from the web, exposing a protected route (according to an env configuration)

By the way, I'm considering extending the package instead of adding too many details to `about`  command. I am waiting for additional feedback.

**Comment by roberto-butti** [comment, marked as answer]
After some discovery, the AboutCommand is quite extendible so I can close this discussion.
If I need to extend the " about" information with additional info, I noticed that I could add info and Sections.
To add info to an existent section like Environment:
```
AboutCommand::add('Environment', "Additional data", "Another value");
```

Creating a new section:
```
AboutCommand::add('LaraSock',
            fn () =>
            [
                'Version' => \Composer\InstalledVersions::getPrettyVersion("hi-folks/lara-sock"),
                'WebSocket engine' => $engine,
                'Number clients connected' => "<fg=red;options=bold>{$connectedClients}</>",
                "Peek num clients" => "<fg=white;options=bold>{$peekclients}</>"
        ]);
```

thank you
